libsigopt/compute/domain.py

```diff
@@ -878,8 +936,9 @@ def rel_prob_func(z):
           one_hot_indexes, categories = zip(*this_cat_dim_map["input_ind_value_map"].items())
           values = one_hot_point[numpy.array(one_hot_indexes, dtype=int)]
           this_cat_point[this_cat_dim_map["output_ind"]] = numpy.random.choice(categories, p=rel_prob_func(values))
-      assert not any(p is None for p in this_cat_point)
-      categorical_points.append(this_cat_point)
+      not_none_points = [p for p in this_cat_point if p is not None]
+      assert len(not_none_points) == self.dim
+      categorical_points.append(not_none_points)
```

Comment:
the variable name needs to be in the singular here, something with `point`. it can be:
```
categorical_point = [d for d in this_cat_point if d is not None] 
assert len(categorical_point) == self.dim
```